vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -400,6 +403,45 @@ func (e *ProposalTxExecutor) RewardValidatorTx(tx *txs.RewardValidatorTx) error
 
 			e.OnCommitState.AddUTXO(utxo)
 			e.OnCommitState.AddRewardUTXO(tx.TxID, utxo)
+
+			offset++
+		}
+
+		// Provide the accrued delegatee rewards from successful delegations.
+		delegateeReward, err := e.OnCommitState.GetDelegateeReward(
+			stakerToRemove.SubnetID,
+			stakerToRemove.NodeID,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to fetch accrued delegatee rewards: %w", err)
+		}
+
+		if delegateeReward > 0 {
+			delegationRewardsOwner := uStakerTx.DelegationRewardsOwner()
+			outIntf, err := e.Fx.CreateOutput(delegateeReward, delegationRewardsOwner)
+			if err != nil {
+				return fmt.Errorf("failed to create output: %w", err)
+			}
+			out, ok := outIntf.(verify.State)
+			if !ok {
+				return errInvalidState
+			}
+			utxo := &avax.UTXO{
+				UTXOID: avax.UTXOID{
+					TxID:        tx.TxID,
+					OutputIndex: uint32(len(outputs) + len(stake) + offset),
+				},
+				Asset: stakeAsset,
+				Out:   out,
+			}
+
+			e.OnCommitState.AddUTXO(utxo)
+			e.OnCommitState.AddRewardUTXO(tx.TxID, utxo)
```

Comment:
Because the delegations have already been earned in this case. We should still reward the validator in the `OnAbortState`. Sorry for the whiplash on the expected behavior here. This should mean that we can get rid of the `rewardAmount` var tho and go back to just using `stakerToRemove.PotentialReward` like we were before.

Reply:
[Whiplash](https://www.imdb.com/title/tt2582802/) is a very good movie

[23f6c87](https://github.com/ava-labs/avalanchego/pull/1262/commits/23f6c87012c182a054488a6c7428c6a05396f07c)